draft-bbs-signatures.md

```diff
@@ -498,22 +496,21 @@ Procedure:
 
 ### ProofGen
 
-This operation computes a zero-knowledge proof-of-knowledge of a signature, while optionally selectively disclosing from the original set of signed messages. The "prover" may also supply a presentation header, see [presentation header selection](#presentation-header-selection) for more details.
+This operation computes a zero-knowledge proof-of-knowledge of a signature, while optionally selectively disclosing from the original set of signed messages. The "prover" may also supply a presentation header, see [presentation header selection](#presentation-header-selection) for more details. 
 
-If an application chooses to pass the indexes of the generators instead, then it will also need to pass the indexes of the generators corresponding to the revealed messages.
+The generators and messages are supplied as pairs, along with a Boolean variable indicating if that message will be revealed or not. The generators and messages MUST be the ones supplied to the Sign operation.
 
 ```
-proof = ProofGen(PK, signature, header, ph, (msg_1,..., msg_L), (H_1,..., H_L), RevealedIndexes)
+proof = ProofGen(PK, signature, header, ph,
+                 [(H_1, msg_1, revealed_1), ..., (H_L, msg_L, revealed_L)])
 
 Inputs:
 
 - PK (REQUIRED), an octet string of the form outputted by the SkToPk operation.
 - signature (REQUIRED), an octet string of the form outputted by the Sign operation.
 - header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
 - ph (OPTIONAL), octet string.
-- msg_1,..., msg_L (OPTIONAL), octet strings. Messages in input to Sign.
-- H_1,..., H_L (OPTIONAL), points of G1. The generators in input to Sign.
-- RevealedIndexes (OPTIONAL), vector of unsigned integers. Indexes of revealed messages.
+- [(H_1, msg_1, revealed_1), ..., (H_L, msg_L, revealed_L)] (OPTIONAL), a list of tuples consisting by a generator, the message and a variable indicating if the message will be revealed or not. Generators must be G1 points, messages must be non-zero scalars mod r and the variable indicating if that message will be revealed must be boolean. If not supplied, the list will default to the empty array ("[]"). The list MUST be ordered based on the index of the generators.
```

Comment:
```suggestion
- [(H_1, msg_1, revealed_1), ..., (H_L, msg_L, revealed_L)] (OPTIONAL), a list of tuples consisting of a generator, the message and a boolean value indicating if the message will be revealed or not. Generators must be G1 points, messages must be non-zero scalars mod r and the variable indicating if that message will be revealed must be boolean. If not supplied, the list will default to the empty array ("[]"). The list MUST be ordered based on the index of the generators.
```
```suggestion
- [(H_1, msg_1, revealed_1), ..., (H_L, msg_L, revealed_L)] (OPTIONAL), a list of tuples consisting by a generator, the message and a variable indicating if the message will be revealed or not. Generators must be G1 points, messages must be non-zero scalars mod r and the variable indicating if that message will be revealed must be boolean. If not supplied, the list will default to the empty array ("[]"). The list MUST be ordered based on the index of the generators.
```

Reply:
The last statement here around the ordering I still dont think is 100% clear, but I think in general this PR is a great improvement.

Reply:
The last statement here around the ordering I still dont think is 100% clear, but I think in general this PR is a great improvement.

Reply:
Changed the statement a bit (now the ordering needs to follow the ordering of the input to Sign) and provided some examples to make it more clear.